bundles/org.openhab.binding.ecovacs/src/main/java/org/openhab/binding/ecovacs/internal/handler/EcovacsVacuumHandler.java

```diff
@@ -0,0 +1,830 @@
+                // ensure we have a mapping (should always be the case)
+                .filter(Optional::isPresent).map(opt -> opt.get())
+                // apply supplied filter
+                .filter(mv -> filter == null || filter.test(mv))
+                // apply capability filter
+                .filter(mv -> mv.capability.isEmpty() || device.hasCapability(mv.capability.get()))
+                // map to actual option
+                .map(mv -> new StateOption(mv.value, mv.value)).collect(Collectors.toList());
+    }
+
+    private synchronized void scheduleNextPoll(long initialDelaySeconds) {
+        final EcovacsVacuumConfiguration config = getConfigAs(EcovacsVacuumConfiguration.class);
+        final long delayUntilNextPoll;
+        if (initialDelaySeconds < 0) {
+            long intervalSeconds = config.refresh * 60;
+            long secondsSinceLastPoll = (System.currentTimeMillis() - lastSuccessfulPollTimestamp) / 1000;
+            long deltaRemaining = intervalSeconds - secondsSinceLastPoll;
+            delayUntilNextPoll = Math.max(0, deltaRemaining);
+        } else {
+            delayUntilNextPoll = initialDelaySeconds;
+        }
+        logger.debug("{}: Scheduling next poll in {}s, refresh interval {}min", getDeviceSerial(), delayUntilNextPoll,
+                config.refresh);
+        pollTask.cancel();
+        pollTask.schedule(delayUntilNextPoll);
+    }
+
+    private void initDevice() {
+        final String serial = getDeviceSerial();
+        if (serial.isEmpty()) {
+            return;
+        }
+
+        final EcovacsApiHandler handler = getApiHandler();
+        if (handler == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_UNINITIALIZED);
+            return;
+        }
+
+        try {
+            final EcovacsApi api = handler.createApiForDevice(serial);
+            api.loginAndGetAccessToken();
+            Optional<EcovacsDevice> deviceOpt = api.getDevices().stream()
+                    .filter(d -> serial.equals(d.getSerialNumber())).findFirst();
+            if (deviceOpt.isPresent()) {
+                EcovacsDevice device = deviceOpt.get();
+                this.device = device;
+                updateProperty(Thing.PROPERTY_MODEL_ID, device.getModelName());
+                updateProperty(Thing.PROPERTY_SERIAL_NUMBER, device.getSerialNumber());
+                updateStateOptions(device);
+                removeUnsupportedChannels(device);
+                connectToDevice();
+            } else {
+                logger.info("{}: Device not found in device list, setting offline", serial);
+                updateStatus(ThingStatus.OFFLINE);
+            }
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        } catch (ConfigurationException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
```

Comment:
Does `ConfigurationException` contain a reason that could be used for the status update?